### PR TITLE
Build worker without dependency on legacy portability core objects

### PR DESCRIPTION
**Pull request by holachuy**
Also removed storage of private key

### CODE REVIEW

File: portability-worker/build.gradle [hunk 1]
```diff
@@ -47,8 +47,11 @@ configurations {
 }
 
 dependencies {
-    compile project(':portability-core')
+    compile project(':portability-spi-cloud')
     compile project(':portability-spi-transfer')
+    compile project(':libraries:security')
+    // TODO: Make this test compile only
```
Comment (rtannenbaum) [inline portability-worker/build.gradle]: done, right? can you move to the testCompile section?

**Comment by holachuy** [comment]
This should be updated and rebased against previous reformatting changes.

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java [hunk 2]
```diff
@@ -46,11 +45,14 @@
     private final Logger logger = LoggerFactory.getLogger(JobPollingService.class);
     private final JobStore store;
     private final WorkerJobMetadata jobMetadata;
+    private final AsymmetricKeyGenerator aymmetricKeyGenerator;
 
     @Inject
-    JobPollingService(CloudFactory cloudFactory, WorkerJobMetadata jobMetadata) {
-        this.store = cloudFactory.getJobStore();
+    JobPollingService(JobStore store, WorkerJobMetadata jobMetadata,
+                      AsymmetricKeyGenerator aymmetricKeyGenerator) {
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: nit: typo in variable name

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java [hunk 3]
```diff
@@ -103,19 +105,19 @@ private void pollForUnassignedJob() throws IOException {
...
-    private void keyGenerated(UUID jobId, PublicKey publicKey, PrivateKey privateKey)
+    private void keyGenerated(UUID jobId, PublicKey publicKey)
             throws IOException {
         // Lookup the job so we can append to its existing properties.
         PortabilityJob existingJob = store.findJob(jobId);
         // Verify no worker key
         // Populate job with keys to persist
-        String encodedPublicKey = PublicPrivateKeyPairGenerator.encodeKey(publicKey);
-        String encodedPrivateKey = PublicPrivateKeyPairGenerator.encodeKey(privateKey);
+        Preconditions.checkArgument(existingJob.jobAuthorization().encryptedPublicKey() == null);
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: can you add a message, worker key should not be present in the PortabilityJob yet, as we're about to set it to the key we generated just now

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java [hunk 4]
```diff
@@ -103,19 +105,19 @@ private void pollForUnassignedJob() throws IOException {
...
+        // Populate job with public key to persist
+        String encodedPublicKey = BaseEncoding.base64Url().encode(publicKey.getEncoded());
 
         PortabilityJob updatedJob = existingJob.toBuilder()
                 .setAndValidateJobAuthorization(existingJob.jobAuthorization().toBuilder()
                         .setEncryptedPublicKey(encodedPublicKey)
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: not related to this PR but while you're in here, can you change this to 'setEncodedPublicKey'?

actually, does it even still need to be encoded? was that just so we could pass to the client as a backup, and are we still doing that?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java [hunk 5]
```diff
@@ -37,95 +37,99 @@
...
-              new ExportInformation(null, resource));
-        }
-      }
+    private static final AtomicInteger COPY_ITERATION_COUNTER = new AtomicInteger();
+    private static final Logger logger = LoggerFactory
+            .getLogger(PortabilityInMemoryTransferCopier.class);
+
+    private final TransferServiceProviderRegistry registry;
+
+    public PortabilityInMemoryTransferCopier(TransferServiceProviderRegistry registry) {
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: this one is mostly indent changes right? is the new ctor the only change?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java [hunk 6]
```diff
@@ -83,29 +80,31 @@ private void pollForJob() {
...
                     decrypter.decrypt(jobAuthorization.encryptedExportAuthData());
             AuthData exportAuthData = deSerialize(serializedExportAuthData);
             String serializedImportAuthData =
                     decrypter.decrypt(jobAuthorization.encryptedImportAuthData());
             AuthData importAuthData = deSerialize(serializedImportAuthData);
-            PortabilityCopier
-                    .copyDataType(registry, dataType, job.exportService(), exportAuthData,
-                            job.importService(), importAuthData, jobId);
+            // TODO: Refactor copy logic
+            InMemoryTransferCopier copier = new PortabilityInMemoryTransferCopier(registry);
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: Can copier be injected? (once its ctor is annotated with @Inject)

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java [hunk 7]
```diff
@@ -37,95 +37,99 @@
...
+     *
+     * @param dataType       The data type to transfer
+     * @param exportService  The export service to transfer from
+     * @param exportAuthData The auth data for the export service
+     * @param importService  The import service to transfer to
+     * @param importAuthData The auth data for the import service
+     * @param jobId          The job id representing this copy
+     */
+    @Override
+    public void copyDataType(String dataType,
```
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: A future refactor to consider, but I think would be difficult in the current intermediate state with some copying still being done in the API etc.

Each worker instance should be spun up with one (eventual) definition of all these properties (export service, auth data, job ID, etc) although they will not be available until a job is polled. WDYT of injecting all these fields with providers in the copier, so they are only evaluated once we start the copy, at which point the job is polled?

I think we should follow the same model for local worker testing too - spin up/tear down for each job tested locally

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java (hunk 5 above)
Comment (rtannenbaum) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: @Inject?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java (hunk 2 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: Done

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java (hunk 3 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: Done

File: portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java (hunk 4 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java]: Done. At some point, we will

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java (hunk 5 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: Yes, sorry, that's the only change.

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java (hunk 7 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: Yes, we can refactor as to how to enforce the singleton nature of the data we want to pass around. Maybe an singleton annotated object with these params or individual data.

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java (hunk 6 above)
Comment (holachuy) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: Yes, the TODO should account for that.

File: portability-worker/build.gradle (hunk 1 above)
Comment (rtannenbaum) [reply portability-worker/build.gradle]: Did you see this one?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java (hunk 5 above)
Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: Did you see this one?

Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java]: ok cool, thanks

**Comment by holachuy** [comment]
Still working on why the test dependency doesn't pick up

> Task :portability-worker:compileTestJava FAILED
/home/chuy/Desktop/wt/data-portability/portability-worker/src/test/java/org/dataportabilityproject/worker/JobPollingServiceTest.java:18: error: cannot find symbol
import org.dataportabilityproject.cloud.local.InMemoryKeyValueStore;
                                             ^
  symbol:   class InMemoryKeyValueStore
  location: package org.dataportabilityproject.cloud.local
1 error

**Pull request merged**